src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -221,13 +221,26 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::ProfileDe
 // - a reference to the new profile
 winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::CreateNewProfile()
 {
-    auto newProfile{ _userDefaultProfileSettings->CreateChild() };
-    _allProfiles.Append(*newProfile);
+    if (_allProfiles.Size() == std::numeric_limits<uint32_t>::max())
+    {
+        // Shouldn't really happen
+        return nullptr;
+    }
 
-    // Give the new profile a distinct name so a guid is properly generated
-    const winrt::hstring newName{ fmt::format(L"Profile {}", _allProfiles.Size()) };
-    newProfile->Name(newName);
+    winrt::hstring newName{};
+    for (uint32_t candidateIndex = 0; candidateIndex < _allProfiles.Size() + 1; candidateIndex++)

```

Comment:
nit: should we just start this at `_allProfiles.Size()`? I know we'll never have `UINT32_MAX - Size()` profiles, but won't the additional math on line 234 cause it to overflow then?

Reply:
Starting here from 0 to actually allow overflow (and to wrap around to Profile 0 and so on). This actually ensures that a name is allocated no matter what.

Reply:
CLEVER.